geopandas/io/tests/test_file.py

```diff
@@ -216,6 +216,34 @@ def test_to_file_schema(tmpdir, df_nybb):
     assert result_schema == schema
 
 
+def test_append_file(tmpdir, df_nybb, df_null):
+    """ Test to_file with append mode and from_file """
+    tempfilename = os.path.join(str(tmpdir), "boros.shp")
+    df_nybb.to_file(tempfilename, driver="ESRI Shapefile")
+    df_nybb.to_file(tempfilename, mode="a", driver="ESRI Shapefile")
+    # Read layer back in
+    df = GeoDataFrame.from_file(tempfilename)
+    assert "geometry" in df
+    assert len(df) == (5 * 2)
+    assert np.alltrue(
+        df["BoroName"].values
+        == np.concatenate([df_nybb["BoroName"].values, df_nybb["BoroName"].values])
+    )
```

Comment:
I think you could also do something like

```
expected = pd.concat([df_nybb]*2, ignore_index=True)
assert_geodataframe_equal(df, expected)
```

Reply:
I think I saw that pandas wasn't already imported in the test file, so I searched for a numpy solution... But yes, I agree the pandas way is cleaner